src/view/point-edit-view.js

```diff
@@ -213,19 +220,95 @@ export default class PointEditView extends AbstractStatefulView {
       offers: this.#offers.find((offer) => offer.type.toLowerCase() === this._state.point.type.toLowerCase())?.offers
     };
     return createPointEditTemplate(state, this.#point, this.#destinations, this.#offers);
+  }
+
+  //перегружаем метод родителя removeElement,
+  //чтобы при удалении удалялся более не нужный календарь
+  removeElement() {
+    super.removeElement();
 
+    if (this.#datepickerStart) {
+      this.#datepickerStart.destroy();
+      this.#datepickerStart = null;
+    }
+    if (this.#datepickerEnd) {
+      this.#datepickerEnd.destroy();
+      this.#datepickerEnd = null;
+    }
   }
 
-  /*#formSubmitHandler = (evt) => {
-    evt.preventDefault();
-    this.#handleFormSubmit(this.#point);
-  };*/
+
+  #dueDateChangeHandler = ([userDate]) => {
+    this.updateElement ({
+      dueDate:  userDate,
+    });
+  };
 
   #editClickHandler = (evt) => {
     evt.preventDefault();
     this.#handleEditClick();
   };
 
+  #setDatepickers = () => {
+    const [dateStartElement, dateEndElement] = this.element.querySelectorAll('.event__input--time');
+    this.#datepickerStart = flatpickr(
+      dateStartElement,
+      {
+        dateFormat: 'd/m/y H:i',
+        defaultDate: this._state.startDate,
+        onClose: this.#dateStartChangeHandler,
+        enableTime: true,
+        maxDate: this._state.endDate,
+        locale: {
+          firstDayOfWeek: 1
+        },
+        'time_24hr': true
+      }
+    );
+    this.#datepickerEnd = flatpickr(
+      dateEndElement,
+      {
+        dateFormat: 'd/m/y H:i',
+        defaultDate: this._state.endDate,
+        onClose: this.#dateEndChangeHandler,
+        enableTime: true,
+        minDate: this._state.startDate,
+        locale: {
+          firstDayOfWeek: 1
+        },
+        'time_24hr': true
+      }
+    );
+  };
+
+  #dateStartChangeHandler = ([userDate]) => {
+    this._setState({
+      startDate: userDate,
+    });
+  };
```

Comment:
this._setState({
      point: {
        ...this._state.point,
        dateStart: userDate
      }
    });